### PR TITLE
Introduce additional requirements for the spaces install

### PR DESCRIPTION
**Pull request by tnthornton**
### Description of your changes
- providers (and related components) are required and will be installed
- ingress-nginx is now required and will be installed

This removes some scope creep that was occurring with the `Spaces` install.

I have:

- [X] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

https://github.com/upbound/up/assets/2375126/c8c1729d-1aa6-43a3-b641-6c14c7956897

### CODE REVIEW

File: cmd/up/space/prerequisites/providers/helm/helm.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+		if pkg.GetInstalled() && pkg.GetHealthy() {
+			break
+		}
+	}
+
+	for {
+		if _, err := h.crdclient.
+			CustomResourceDefinitions().
+			Get(
+				context.Background(),
```
Comment (branden) [inline cmd/up/space/prerequisites/providers/helm/helm.go:172]: Should this be reusing the existing context?

File: cmd/up/space/prerequisites/providers/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+		if pkg.GetInstalled() && pkg.GetHealthy() {
+			break
+		}
+	}
+
+	for {
+		if _, err := k.crdclient.
+			CustomResourceDefinitions().
+			Get(
+				context.Background(),
```
Comment (branden) [inline cmd/up/space/prerequisites/providers/kubernetes/kubernetes.go:172]: Should this be reusing the existing context?

**Review by branden — COMMENTED** [review]
I found a couple spots where it looks like we're creating a new context when we should be reusing an existing one. Otherwise LGTM.

File: cmd/up/space/prerequisites/providers/helm/helm.go (hunk 1 above)
Comment (tnthornton) [reply cmd/up/space/prerequisites/providers/helm/helm.go]: Fixed.

File: cmd/up/space/prerequisites/providers/kubernetes/kubernetes.go (hunk 2 above)
Comment (tnthornton) [reply cmd/up/space/prerequisites/providers/kubernetes/kubernetes.go]: Fixed.

**Review by branden — APPROVED** [review]

**Pull request merged**